chart/templates/jobs/migrate-database-job.yaml

```diff
@@ -38,6 +38,9 @@ metadata:
     "helm.sh/hook": post-install,post-upgrade
     "helm.sh/hook-weight": "1"
     "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+    "argocd.argoproj.io/hook": Sync
+    "argocd.argoproj.io/sync-wave": "0"
+    "argocd.argoproj.io/hook-delete-policy": BeforeHookCreation,HookSucceeded
```

Comment:
Would it help to allow adding extra annotations as part of `values.yaml`, kind of similar to https://github.com/apache/airflow/blob/bff213e07735d1ee45101f85b01b3d3a97cddbe5/chart/templates/configmaps/configmap.yaml#L34-L37 

?

cc @jedcunningham

Reply:
Yeah, I'd prefer to expose extraAnnotations for these jobs rather than add in specific annotations for a specific platform. @dnskr, something you want to tackle?

Reply:
Yes, it looks like reasonable place to add my workaround. So in the future there is no need to remove annotations from Airflow codebase. This PR at least will help people to find the solution for the problem I hope.

@jedcunningham Do you have time to implement this? Otherwise I'll do it later, probably on a weekend.

Reply:
Yeah, I'll get a PR in today, no worries. Thanks for the detailed explanation btw!

@kaxil, I feel like we should toss this into the docs somewhere, maybe a new page with platform/tool-specific notes? Beats expecting folks to hit google and sift through PRs. Let me know if you have a preference on naming.

Reply:
Oh, looks like I already added this in #16058. It will be in the next chart release 👍.

Reply:
Yeah, having docs for that would be nice, similar to https://github.com/airflow-helm/charts/tree/main/charts/airflow/examples but in docs

Reply:
Cool! Looking forward for the next release :)

Reply:
@jedcunningham In the PR [#16058](https://github.com/apache/airflow/pull/16058) you have added annotations on Pod level, but they are needed on Job level.
So the same as [this](https://github.com/apache/airflow/blob/main/chart/templates/jobs/migrate-database-job.yaml#L53-L55) piece of code but [here](https://github.com/apache/airflow/blob/main/chart/templates/jobs/migrate-database-job.yaml#L37)

Reply:
Ah, good catch, thanks. I'll get that in in the next couple days.